homeassistant/components/utility_meter/sensor.py

```diff
@@ -283,7 +276,7 @@ async def async_reset_meter(self, entity_id):
             return
         _LOGGER.debug("Reset utility meter <%s>", self.entity_id)
         self._last_reset = dt_util.utcnow()
-        self._last_period = str(self._state)
+        self._last_period = str(self._state) if self._state else 0
```

Comment:
We should probably not mix types here. Either set a string or an integer or a float etc.

Reply:
I think this is the place where we convert `None` to `"None"`. I think we should remove the string copy here.
```suggestion
        self._last_period = self._state if self._state else 0
```

Reply:
str is required to convert Decimal into something JSON serialisable without looking into state datatype

Reply:
We convert to `float` when restoring state. We need to decide what type `self._last_period` should be. It can't be one type in one place and another type in another place.

Reply:
made it always a Decimal and cast into str in the attribute only